addons/misra.py

```diff
@@ -3185,6 +3187,38 @@ def misra_17_3(self, cfg):
         for w in cfg.clang_warnings:
             if w['message'].endswith('[-Wimplicit-function-declaration]'):
                 self.reportError(cppcheckdata.Location(w), 17, 3)
+        for token in cfg.tokenlist:
+            if token.str not in ["while", "if"]:
+                continue
+            if token.next.str != "(":
+                continue
+            tok = token.next
+            while (tok != token.next.link):
```

Comment:
will python lookup token.next.link every loop? would it be slightly faster to write:

    end_token = token.next.link
    while tok != end_token: